driven/storage/adapter_client.go

```diff
@@ -9,44 +9,89 @@ import (
 	"go.mongodb.org/mongo-driver/bson"
 )
 
-// GetUserCourses finds user course by a set of parameters
-func (sa *Adapter) GetUserCourses(id []string, name []string, key []string, userID string) ([]model.UserCourse, error) {
-	filter := bson.M{}
+// FindUserCourses finds user courses by the given search parameters
+func (sa *Adapter) FindUserCourses(id []string, appID string, orgID string, name []string, key []string, userID *string, timezoneOffsetPairs []model.TZOffsetPair, requirements map[string]interface{}) ([]model.UserCourse, error) {
+	filter := bson.M{"app_id": appID, "org_id": orgID}
 	if len(id) != 0 {
 		filter["_id"] = bson.M{"$in": id}
 	}
-
 	if len(name) != 0 {
 		filter["course.name"] = bson.M{"$in": name}
 	}
-
 	if len(key) != 0 {
 		filter["course.key"] = bson.M{"$in": key}
 	}
-
-	if len(userID) != 0 {
+	if userID != nil {
 		filter["user_id"] = userID
 	}
 
-	var result []userCourse
-	err := sa.db.userCourses.Find(sa.context, filter, &result, nil)
-	if err != nil {
-		return nil, err
+	// timezone offsets
+	if len(timezoneOffsetPairs) > 0 {
+		offsetFilters := make(bson.A, 0)
+		for _, offsetPair := range timezoneOffsetPairs {
+			offsetFilters = append(offsetFilters,
+				bson.M{
+					"timezone_offset": bson.M{
+						"$gte": offsetPair.Lower,
+						"$lte": offsetPair.Upper,
+					},
+				},
+			)
+		}
+		filter["$or"] = offsetFilters
 	}
-	if len(result) == 0 {
-		//no data
-		return nil, nil
+
+	// notification requirements
+	for reqKey, reqVal := range requirements {
+		if reqKey == "completed_tasks" {
+			now := time.Now()
+			y, m, d := now.Date()
```

Comment:
I think this loop may need to be done in `app_streaks_notifications.go` because `now` needs to be converted to the correct timezone using `.FixedZone` or `.In`. Add a value for `"completed_tasks_time"` in the requirements map, which should be the converted `time.Time`. Use the `"completed_tasks"` boolean as you do here (to determine whether to use `$gte` or `$lt`.